vyper/codegen/module.py

```diff
@@ -100,6 +106,10 @@ def _runtime_ir(runtime_functions, all_sigs, global_ctx):
     internal_functions = [f for f in runtime_functions if _is_internal(f)]
 
     external_functions = [f for f in runtime_functions if not _is_internal(f)]
+
+    # returns True if any function's selector has trailing 0s
+    has_zero_selector_functions = any(_is_zero_selector_func(f) for f in external_functions)
```

Comment:
so i think we need to check actually all possible methods for each function. so something like `ContractFunction.method_ids`.

Reply:
changed with my latest commit - let me know if that's what you were thinking